### ISSUE TITLE
V3 - Enhance Alteration Engine to support specific Activity Instances

### ISSUE THREAD

**Issue by b0rv3g4r4** [elsa-workflows/elsa-core#4644]
**Use case**

We have a flow where it is iterated in parallel for each contracted order. Within each iteration a call is being made to a Rest API to integrate the data. It is possible that the service is down or fails in 1 (or more) of the iterations for the same Activity Id (the Rest call). It would be necessary to be able to retry at the exact point where it failed and continue the execution whether it is a single error or multiple errors.

Perhaps from the instance view we could have a button to relaunch from the specific failed execution or from the incident.

It would also be nice to have a generic retry button as Elsa 2 already suggests in the list of instances to relaunch all the activities in error.

**Expected behavior**

Retry the flow from the exact activity instance that failed in order to correct the problem explained in the use case.

It would also be nice to have a "Retry All" button to cover the case in which several iterations fail due to a service outage.
